uptrends/client.go

```diff
@@ -0,0 +1,68 @@
+package uptrends
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type clientor interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type requestCreator interface {
+	NewRequest(method, path string, body io.Reader) (*http.Request, error)
+}
+
+type httpReqCreator struct{}
+
+func (h *httpReqCreator) NewRequest(method, path string, body io.Reader) (*http.Request, error) {
+	return http.NewRequest(method, path, body)
+}
+
+// Client is an Uptrends.com API Client
+type Client struct {
+	baseURL url.URL
+	clientor
+	username string
+	password string
+	requestCreator
+}
+
+// New returns an Uptrends.com API Client
+//  username is the Uptrends.com login user
+//  password is the corresponding password for username
+func New(username string, password string) (*Client, error) {
+	u, err := url.Parse("https://api.uptrends.com")
```

Comment:
Not sure if uptrends provides a test/staging API endpoint, but you might consider passing the host or base URI in as an optional third parameter and use this URI as the default.

Reply:
Thanks for the feedback, @jesselang .

I originally implemented it as you describe and then I removed that parameter. I don't see where Uptrends.com provides any test API endpoint, but the more I look at this it seems useful to expose this as an input. I'll look at doing that in a future iteration.